src/pages/products/products-nav-filters/products-nav-filters.js

```diff
@@ -0,0 +1,172 @@
+  const { filterData, filters } = useSelector(({ Products }) => ({
+    filterData: Products.filterData,
+    filters: Products.filters
+  }));
+
+  const {
+    categoryFilter,
+    colorsFilter,
+    patternsFilter,
+    modelsFilter
+  } = filters;
+
+  const categoriesNames = useMemo(
+    () => [
+      ...new Set(filterData.map(({ category }) => category.name[0].value))
+    ],
+    [filterData]
+  );
+
+  const categories = useMemo(
+    () =>
+      categoriesNames.map(
+        (category) =>
+          filterData.find(
+            ({ category: { name } }) => category === name[0].value
+          ).category
+      ),
+    [filterData, categoriesNames]
+  );
+
+  const colorsNames = useMemo(
+    () => [
+      ...new Set(filterData.map(({ colors }) => colors[0].simpleName[0].value))
+    ],
+    [filterData]
+  );
+
+  const colors = useMemo(
+    () =>
+      colorsNames.map(
+        (color) =>
+          filterData.find(
+            ({ colors }) => colors[0].simpleName[0].value === color
+          ).colors
+      ),
+    [filterData, colorsNames]
+  );
+
+  const patternsNames = useMemo(
+    () => [...new Set(filterData.map(({ pattern }) => pattern[0].value))],
+    [filterData]
+  );
+
+  const patterns = useMemo(
+    () =>
+      patternsNames.map(
+        (item) =>
+          filterData.find(({ pattern }) => pattern[0].value === item).pattern
+      ),
+    [filterData, patternsNames]
+  );
+
+  const modelNames = useMemo(
+    () => [...new Set(filterData.map(({ model }) => model[0].value))],
+    [filterData]
+  );
+
+  const models = useMemo(
+    () =>
+      modelNames.map(
+        (item) => filterData.find(({ model }) => model[0].value === item).model
+      ),
+    [filterData, modelNames]
+  );
+
+  const handleFilterChange = ({ target }, setFilter) => {
+    dispatch(setFilter(target.value));
+    dispatch(setCurrentPage(0));
+  };
+
+  const filtersOptions = {
+    categories: {
+      buttonName: CATEGORIES,
+      productFilter: categoryFilter,
+      list: categories,
+      labels: categoriesNames,
+      filterHandler: (e) => handleFilterChange(e, setCategoryFilter)
+    },
+    models: {
+      buttonName: MODELS,
+      productFilter: modelsFilter,
+      list: models,
+      labels: modelNames,
+      filterHandler: (e) => handleFilterChange(e, setModelsFilter)
+    },
+    colors: {
+      buttonName: COLORS,
+      productFilter: colorsFilter,
+      list: colors,
+      labels: colorsNames,
+      filterHandler: (e) => handleFilterChange(e, setColorsFilter)
+    },
+    patterns: {
+      buttonName: PATTERNS,
+      productFilter: patternsFilter,
+      list: patterns,
+      labels: patternsNames,
+      filterHandler: (e) => handleFilterChange(e, setPatternsFilter)
+    }
+  };
+
+  const filterButtons = Object.values(
+    filtersOptions
+  ).map(({ buttonName, productFilter, list, labels, filterHandler }, idx) => (
+    <ProductsFilterContainer
+      key={labels[idx]}
```

Comment:
It is the same issue as if you use `idx` directly because `labesl[idx]` depends on the `idx`, please, use some unique value from the filter object but without depending on the `idx`

Reply:
fixed